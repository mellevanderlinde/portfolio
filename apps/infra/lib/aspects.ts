import type { IAspect } from 'aws-cdk-lib'
import type { IConstruct } from 'constructs'
import { CfnResource, RemovalPolicy } from 'aws-cdk-lib'

export class RemovalPolicyDestroyAspect implements IAspect {
  visit(node: IConstruct): void {
    if (CfnResource.isCfnResource(node)) {
      node.applyRemovalPolicy(RemovalPolicy.DESTROY)
    }
  }
}
