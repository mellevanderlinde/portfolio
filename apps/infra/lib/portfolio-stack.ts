import {
  Duration,
  RemovalPolicy,
  Stack,
  type StackProps,
  aws_certificatemanager as certificatemanager,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins as cloudfront_origins,
  aws_logs as logs,
  aws_route53 as route53,
  aws_route53_targets as route53_targets,
  aws_s3 as s3,
  aws_s3_deployment as s3_deployment,
} from "aws-cdk-lib";
import { NagSuppressions } from "cdk-nag";
import type { Construct } from "constructs";

export class PortfolioStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const domainName = "mellevanderlinde.com";
    const wwwDomainName = `www.${domainName}`;
    const zone = route53.HostedZone.fromLookup(this, "HostedZone", {
      domainName,
    });

    const bucket = new s3.Bucket(this, "Bucket", {
      bucketName: domainName,
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
      enforceSSL: true,
      removalPolicy: RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    });

    NagSuppressions.addResourceSuppressions(bucket, [
      {
        id: "AwsSolutions-S1",
        reason: "Server access logging is not required",
      },
    ]);

    const certificate = new certificatemanager.Certificate(
      this,
      "Certificate",
      {
        certificateName: domainName,
        domainName,
        validation: certificatemanager.CertificateValidation.fromDns(zone),
        subjectAlternativeNames: [wwwDomainName],
      },
    );

    const function_ = new cloudfront.Function(this, "Function", {
      functionName: domainName.replace(".", "-"),
      code: cloudfront.FunctionCode.fromFile({ filePath: "src/index.js" }),
      runtime: cloudfront.FunctionRuntime.JS_2_0,
      comment: "Add index.html to URI (required for Next.js)",
    });

    const distribution = new cloudfront.Distribution(this, "Distribution", {
      defaultRootObject: "index.html",
      defaultBehavior: {
        origin:
          cloudfront_origins.S3BucketOrigin.withOriginAccessControl(bucket),
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        functionAssociations: [
          {
            function: function_,
            eventType: cloudfront.FunctionEventType.VIEWER_REQUEST,
          },
        ],
        responseHeadersPolicy:
          cloudfront.ResponseHeadersPolicy.SECURITY_HEADERS,
      },
      priceClass: cloudfront.PriceClass.PRICE_CLASS_100,
      httpVersion: cloudfront.HttpVersion.HTTP3,
      domainNames: [domainName, wwwDomainName],
      enableIpv6: true,
      certificate,
      errorResponses: [
        {
          httpStatus: 403,
          responseHttpStatus: 404,
          responsePagePath: "/404.html",
          ttl: Duration.minutes(30),
        },
      ],
      geoRestriction: cloudfront.GeoRestriction.allowlist(
        "AT",
        "BE",
        "CH",
        "DE",
        "DK",
        "ES",
        "FR",
        "GB",
        "IE",
        "IT",
        "LU",
        "NL",
        "PL",
        "PT",
        "US",
      ),
    });

    NagSuppressions.addResourceSuppressions(distribution, [
      {
        id: "AwsSolutions-CFR2",
        reason: "AWS WAF is not required",
      },
      {
        id: "AwsSolutions-CFR3",
        reason: "Access logging is not required",
      },
    ]);

    const target = route53.RecordTarget.fromAlias(
      new route53_targets.CloudFrontTarget(distribution),
    );

    new route53.ARecord(this, "ARecord", {
      recordName: domainName,
      target,
      zone,
    });

    new route53.ARecord(this, "WwwARecord", {
      recordName: wwwDomainName,
      target,
      zone,
    });

    new route53.AaaaRecord(this, "AaaaRecord", {
      recordName: domainName,
      target,
      zone,
    });

    new route53.AaaaRecord(this, "WwwAaaaRecord", {
      recordName: wwwDomainName,
      target,
      zone,
    });

    new route53.TxtRecord(this, "TxtRecord", {
      values: [
        "google-site-verification=cdnxZVubxHlWGwUWIl8O3zQA57brfgrjMC-gzOibbEQ",
      ],
      zone,
    });

    const logGroup = new logs.LogGroup(this, "LogGroup", {
      logGroupName: domainName,
      retention: logs.RetentionDays.ONE_DAY,
    });

    const bucketDeployment = new s3_deployment.BucketDeployment(
      this,
      "BucketDeployment",
      {
        destinationBucket: bucket,
        sources: [s3_deployment.Source.asset("../website/out")],
        distribution,
        distributionPaths: ["/*"],
        logGroup,
        memoryLimit: 2048,
      },
    );

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      "/PortfolioStack/Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiB/Resource",
      [
        {
          id: "AwsSolutions-L1",
          reason: "This Lambda is managed by CDK",
        },
      ],
    );

    NagSuppressions.addResourceSuppressions(
      bucketDeployment.handlerRole,
      [
        {
          id: "AwsSolutions-IAM5",
          reason: "Full S3 access is required",
        },
        {
          id: "AwsSolutions-IAM4",
          reason: "Write access to CloudWatch is allowed",
        },
      ],
      true,
    );
  }
}
