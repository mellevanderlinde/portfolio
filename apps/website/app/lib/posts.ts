import fs from 'node:fs'
import path from 'node:path'

interface Metadata {
  title: string
  publishedAt: string
}

interface BlogPost {
  metadata: Metadata
  slug: string
  content: string
}

function parseFrontmatter(fileContent: string): Omit<BlogPost, 'slug'> {
  // eslint-disable-next-line regexp/no-super-linear-backtracking
  const frontmatterRegex = /---\s*([\s\S]*?)\s*---/
  const match = frontmatterRegex.exec(fileContent)
  if (!match) {
    throw new Error('Frontmatter not found')
  }
  const frontMatterBlock = match[1]
  const content = fileContent.replace(frontmatterRegex, '').trim()
  const frontMatterLines = frontMatterBlock.trim().split('\n')
  const metadata: Partial<Metadata> = {}

  for (const line of frontMatterLines) {
    const [key, ...valueArr] = line.split(': ')
    let value = valueArr.join(': ').trim()
    value = value.replace(/^['"](.*)['"]$/, '$1')
    metadata[key.trim() as keyof Metadata] = value
  }

  return { metadata: metadata as Metadata, content }
}

function getMDXFiles(dir: string): string[] {
  return fs.readdirSync(dir).filter(file => path.extname(file) === '.mdx')
}

function readMDXFile(filePath: string): Omit<BlogPost, 'slug'> {
  const rawContent = fs.readFileSync(filePath, 'utf-8')
  return parseFrontmatter(rawContent)
}

function getMDXData(dir: string): BlogPost[] {
  const mdxFiles = getMDXFiles(dir)
  return mdxFiles.map((file) => {
    const { metadata, content } = readMDXFile(path.join(dir, file))
    const slug = path.basename(file, path.extname(file))
    return {
      metadata,
      slug,
      content,
    }
  })
}

export function getBlogPosts(): BlogPost[] {
  // eslint-disable-next-line node/prefer-global/process
  return getMDXData(path.join(process.cwd(), 'content'))
}

export function formatDate(date: string): string {
  return new Date(date).toLocaleString('en-us', {
    month: 'short',
    day: 'numeric',
    year: 'numeric',
  })
}
