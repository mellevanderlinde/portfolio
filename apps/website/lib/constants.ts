import type { Transition, Variants } from 'motion/react'

export const url = 'https://mellevanderlinde.com'

export const variantsContainer: Variants = {
  hidden: { opacity: 0 },
  visible: {
    opacity: 1,
    transition: {
      staggerChildren: 0.15,
    },
  },
}

export const variantsSection: Variants = {
  hidden: { opacity: 0, y: 20, filter: 'blur(8px)' },
  visible: { opacity: 1, y: 0, filter: 'blur(0px)' },
}

export const transition: Transition = {
  duration: 0.3,
}
