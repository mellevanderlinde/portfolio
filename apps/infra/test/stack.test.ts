import { App } from 'aws-cdk-lib'
import { Template } from 'aws-cdk-lib/assertions'
import { expect, it } from 'vitest'
import { PortfolioStack } from '../lib/portfolio-stack'

it('matches with snapshot', () => {
  const app = new App()
  const stack = new PortfolioStack(app, 'PortfolioStack', {
    env: { region: 'us-east-1', account: '012345678912' },
  })
  const template = Template.fromStack(stack)

  // Remove hashes from snapshot
  const bucketDeployment = template.findResources(
    'Custom::CDKBucketDeployment',
  )
  for (const key of Object.keys(bucketDeployment)) {
    bucketDeployment[key].Properties.SourceObjectKeys = ['removed-hash']
  }
  const lambdaFunction = template.findResources('AWS::Lambda::Function')
  for (const key of Object.keys(lambdaFunction)) {
    lambdaFunction[key].Properties.Code.S3Key = 'removed-hash'
  }
  const layerVersion = template.findResources('AWS::Lambda::LayerVersion')
  for (const key of Object.keys(layerVersion)) {
    layerVersion[key].Properties.Content.S3Key = 'removed-hash'
  }

  expect(template).toMatchSnapshot()
})
