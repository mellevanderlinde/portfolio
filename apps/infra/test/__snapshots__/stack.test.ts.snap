// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches with snapshot 1`] = `
{
  "Mappings": {
    "AWSCloudFrontPartitionHostedZoneIdMap": {
      "aws": {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "ARecordE7B57761": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "Distribution830FAC52",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "mellevanderlinde.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "AaaaRecord61D2056E": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "Distribution830FAC52",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "mellevanderlinde.com.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "Bucket83908E77": {
      "DeletionPolicy": "Delete",
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-S1",
              "reason": "Server access logging is not required",
            },
          ],
        },
      },
      "Properties": {
        "BucketName": "mellevanderlinde.com",
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          {
            "Key": "aws-cdk:cr-owned:a50fa62f",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "BucketAutoDeleteObjectsCustomResourceBAFD23C2": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "BucketPolicyE9A3008A",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "Bucket83908E77",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "BucketDeploymentAwsCliLayerB018DCE1": {
      "Properties": {
        "Content": {
          "S3Bucket": "cdk-hnb659fds-assets-012345678912-us-east-1",
          "S3Key": "removed-hash",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "BucketDeploymentCustomResource2048MiB82BEA142": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "Bucket83908E77",
        },
        "DistributionId": {
          "Ref": "Distribution830FAC52",
        },
        "DistributionPaths": [
          "/*",
        ],
        "OutputObjectKeys": true,
        "Prune": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiB8AA4A604",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          "cdk-hnb659fds-assets-012345678912-us-east-1",
        ],
        "SourceObjectKeys": [
          "removed-hash",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "BucketPolicyE9A3008A": {
      "Properties": {
        "Bucket": {
          "Ref": "Bucket83908E77",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "Bucket83908E77",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "Bucket83908E77",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "Bucket83908E77",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "Bucket83908E77",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Condition": {
                "StringEquals": {
                  "AWS:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":cloudfront::",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":distribution/",
                        {
                          "Ref": "Distribution830FAC52",
                        },
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "cloudfront.amazonaws.com",
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "Bucket83908E77",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "Certificate4E7ABB08": {
      "Properties": {
        "DomainName": "mellevanderlinde.com",
        "DomainValidationOptions": [
          {
            "DomainName": "mellevanderlinde.com",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "www.mellevanderlinde.com",
            "HostedZoneId": "DUMMY",
          },
        ],
        "SubjectAlternativeNames": [
          "www.mellevanderlinde.com",
        ],
        "Tags": [
          {
            "Key": "Name",
            "Value": "mellevanderlinde.com",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiB8AA4A604": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleDefaultPolicy5D765796",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleB37F9ACF",
      ],
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-L1",
              "reason": "This Lambda is managed by CDK",
            },
          ],
        },
      },
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-012345678912-us-east-1",
          "S3Key": "removed-hash",
        },
        "Environment": {
          "Variables": {
            "AWS_CA_BUNDLE": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
          },
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "BucketDeploymentAwsCliLayerB018DCE1",
          },
        ],
        "LoggingConfig": {
          "LogGroup": {
            "Ref": "LogGroupF5B46931",
          },
        },
        "MemorySize": 2048,
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleB37F9ACF",
            "Arn",
          ],
        },
        "Runtime": "python3.11",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleB37F9ACF": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "Full S3 access is required",
            },
            {
              "id": "AwsSolutions-IAM4",
              "reason": "Write access to CloudWatch is allowed",
            },
          ],
        },
      },
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleDefaultPolicy5D765796": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "Full S3 access is required",
            },
            {
              "id": "AwsSolutions-IAM4",
              "reason": "Write access to CloudWatch is allowed",
            },
          ],
        },
      },
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-012345678912-us-east-1",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-012345678912-us-east-1/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "Bucket83908E77",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "Bucket83908E77",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleDefaultPolicy5D765796",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C2048MiBServiceRoleB37F9ACF",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-012345678912-us-east-1",
          "S3Key": "removed-hash",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "Bucket83908E77",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "Distribution830FAC52": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-CFR2",
              "reason": "AWS WAF is not required",
            },
            {
              "id": "AwsSolutions-CFR3",
              "reason": "Access logging is not required",
            },
          ],
        },
      },
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "mellevanderlinde.com",
            "www.mellevanderlinde.com",
          ],
          "CustomErrorResponses": [
            {
              "ErrorCachingMinTTL": 1800,
              "ErrorCode": 403,
              "ResponseCode": 404,
              "ResponsePagePath": "/404.html",
            },
          ],
          "DefaultCacheBehavior": {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "FunctionAssociations": [
              {
                "EventType": "viewer-request",
                "FunctionARN": {
                  "Fn::GetAtt": [
                    "Function76856677",
                    "FunctionARN",
                  ],
                },
              },
            ],
            "ResponseHeadersPolicyId": "67f7725c-6f97-4210-82d7-5512b31e9d03",
            "TargetOriginId": "PortfolioStackDistributionOrigin13DFF474E",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http3",
          "IPV6Enabled": true,
          "Origins": [
            {
              "DomainName": {
                "Fn::GetAtt": [
                  "Bucket83908E77",
                  "RegionalDomainName",
                ],
              },
              "Id": "PortfolioStackDistributionOrigin13DFF474E",
              "OriginAccessControlId": {
                "Fn::GetAtt": [
                  "DistributionOrigin1S3OriginAccessControlEB606076",
                  "Id",
                ],
              },
              "S3OriginConfig": {
                "OriginAccessIdentity": "",
              },
            },
          ],
          "PriceClass": "PriceClass_100",
          "Restrictions": {
            "GeoRestriction": {
              "Locations": [
                "AT",
                "BE",
                "CH",
                "DE",
                "DK",
                "ES",
                "FR",
                "GB",
                "IE",
                "IT",
                "LU",
                "NL",
                "PL",
                "PT",
                "US",
              ],
              "RestrictionType": "whitelist",
            },
          },
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Ref": "Certificate4E7ABB08",
            },
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "DistributionOrigin1S3OriginAccessControlEB606076": {
      "Properties": {
        "OriginAccessControlConfig": {
          "Name": "PortfolioStackDistributionOrigin1S3OriginAccessControl63F8F727",
          "OriginAccessControlOriginType": "s3",
          "SigningBehavior": "always",
          "SigningProtocol": "sigv4",
        },
      },
      "Type": "AWS::CloudFront::OriginAccessControl",
    },
    "Function76856677": {
      "Properties": {
        "AutoPublish": true,
        "FunctionCode": ""use strict";
// eslint-disable-next-line unused-imports/no-unused-vars
async function handler(event) {
    const request = event.request;
    const uri = request.uri;
    // Check whether the URI is missing a file name.
    if (uri.endsWith('/')) {
        request.uri += 'index.html';
    }
    // Check whether the URI is missing a file extension.
    else if (!uri.includes('.')) {
        request.uri += '/index.html';
    }
    return request;
}
// See https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/example-function-add-index.html
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiO0FBUUEseURBQXlEO0FBQ3pELEtBQUssVUFBVSxPQUFPLENBQUMsS0FBWTtJQUNqQyxNQUFNLE9BQU8sR0FBRyxLQUFLLENBQUMsT0FBTyxDQUFBO0lBQzdCLE1BQU0sR0FBRyxHQUFHLE9BQU8sQ0FBQyxHQUFHLENBQUE7SUFFdkIsZ0RBQWdEO0lBQ2hELElBQUksR0FBRyxDQUFDLFFBQVEsQ0FBQyxHQUFHLENBQUMsRUFBRSxDQUFDO1FBQ3RCLE9BQU8sQ0FBQyxHQUFHLElBQUksWUFBWSxDQUFBO0lBQzdCLENBQUM7SUFDRCxxREFBcUQ7U0FDaEQsSUFBSSxDQUFDLEdBQUcsQ0FBQyxRQUFRLENBQUMsR0FBRyxDQUFDLEVBQUUsQ0FBQztRQUM1QixPQUFPLENBQUMsR0FBRyxJQUFJLGFBQWEsQ0FBQTtJQUM5QixDQUFDO0lBRUQsT0FBTyxPQUFPLENBQUE7QUFDaEIsQ0FBQztBQUVELHlHQUF5RyIsInNvdXJjZXNDb250ZW50IjpbImludGVyZmFjZSBSZXF1ZXN0IHtcbiAgdXJpOiBzdHJpbmdcbn1cblxuaW50ZXJmYWNlIEV2ZW50IHtcbiAgcmVxdWVzdDogUmVxdWVzdFxufVxuXG4vLyBlc2xpbnQtZGlzYWJsZS1uZXh0LWxpbmUgdW51c2VkLWltcG9ydHMvbm8tdW51c2VkLXZhcnNcbmFzeW5jIGZ1bmN0aW9uIGhhbmRsZXIoZXZlbnQ6IEV2ZW50KTogUHJvbWlzZTxSZXF1ZXN0PiB7XG4gIGNvbnN0IHJlcXVlc3QgPSBldmVudC5yZXF1ZXN0XG4gIGNvbnN0IHVyaSA9IHJlcXVlc3QudXJpXG5cbiAgLy8gQ2hlY2sgd2hldGhlciB0aGUgVVJJIGlzIG1pc3NpbmcgYSBmaWxlIG5hbWUuXG4gIGlmICh1cmkuZW5kc1dpdGgoJy8nKSkge1xuICAgIHJlcXVlc3QudXJpICs9ICdpbmRleC5odG1sJ1xuICB9XG4gIC8vIENoZWNrIHdoZXRoZXIgdGhlIFVSSSBpcyBtaXNzaW5nIGEgZmlsZSBleHRlbnNpb24uXG4gIGVsc2UgaWYgKCF1cmkuaW5jbHVkZXMoJy4nKSkge1xuICAgIHJlcXVlc3QudXJpICs9ICcvaW5kZXguaHRtbCdcbiAgfVxuXG4gIHJldHVybiByZXF1ZXN0XG59XG5cbi8vIFNlZSBodHRwczovL2RvY3MuYXdzLmFtYXpvbi5jb20vQW1hem9uQ2xvdWRGcm9udC9sYXRlc3QvRGV2ZWxvcGVyR3VpZGUvZXhhbXBsZS1mdW5jdGlvbi1hZGQtaW5kZXguaHRtbFxuIl19",
        "FunctionConfig": {
          "Comment": "Add index.html to URI (required for Next.js)",
          "Runtime": "cloudfront-js-2.0",
        },
        "Name": "mellevanderlinde-com",
      },
      "Type": "AWS::CloudFront::Function",
    },
    "LogGroupF5B46931": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "LogGroupName": "mellevanderlinde.com",
        "RetentionInDays": 1,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "TxtRecordCFF063CA": {
      "Properties": {
        "HostedZoneId": "DUMMY",
        "Name": "mellevanderlinde.com.",
        "ResourceRecords": [
          ""google-site-verification=cdnxZVubxHlWGwUWIl8O3zQA57brfgrjMC-gzOibbEQ"",
        ],
        "TTL": "1800",
        "Type": "TXT",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "WwwARecord21A54661": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "Distribution830FAC52",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.mellevanderlinde.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "WwwAaaaRecord7E37C242": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "Distribution830FAC52",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.mellevanderlinde.com.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
